.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+on: pull_request
+jobs:
+  pre-commit:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Setup python 3.6
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+  test-unit:
+    runs-on: ubuntu-18.04
+    strategy:
+        matrix:
+          python-version: ['3.6.x', '3.7.x']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Setup python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: pip3 install -r requirements.txt && pip3 install .[all] && pip3 install codecov
+      - name: Run python unit tests
+        run: make test
```

Comment:
I think we miss here running the FE tests and a build.
I will check it tomorrow and add a suggestion!

Reply:
Maybe something like:
```
- name: Setup Frontend Application
  run: cd amundsen_application/static
- name: Install Frontend Dependencies
  run: npm install 
  run: npm install codecov -g
- name: Lint Frontend
  run: npm run lint
- name: Build Frontend
  run: npm run build
- name: Test Frontend
  run: npm run test
- name: Record Test Coverage for Frontend
  run: codecov
```

Reply:
thanks @Golodhros , pr updated.